### DISCUSSION TITLE
Determine if a provider has a value

### DISCUSSION THREAD

**Discussion by scopendo** [rrousselGit/riverpod#3356] [Q&A] [answered]
Hi – my app has a provider that contains a list of bookings.  I also have a booking notifier that allows operations on a single booking.  When initialising the booking notifier, I want to use the value from the list of bookings if it's in there, whilst loading the data for the booking asynchronously.  This allows the UI to show something more quickly and then to update if the data was stale.

```dart
class BookingNotifier extends AutoDisposeFamilyAsyncNotifier<BookingInfo, int> {
  @override
  Future<BookingInfo> build(int arg) async {
    // See if we can find the booking in the list of current bookings
    final bookings = await ref.watch(Pods.currentBookings.future);
    final bi = bookings.firstWhereOrNull((bi) => bi.booking.id == arg);

    if (bi == null) {
      // We'll need to wait for the data to load
      return _loadBooking(arg);
    } else {
      // We can use the data we've got in the booking object whilst we wait
      // for the async data load to complete
      _loadBooking(arg)
          .then((bookingInfo) => state = AsyncData(bookingInfo))
          .withoutAwait();
      return bi;
    }
  }
  ...
}
```

This works fine, however, it will force provider `Pods.currentBookings` to load if it hasn't loaded already.

Is there a way to check whether a provider is already loaded without triggering a load?

**Comment by rrousselGit** [comment, marked as answer]
Are you looking for `ref.exist(provider)`?

**Reply by scopendo** [reply]
Yes – I was looking for something like that but was clearly looking in the place.  Thanks @rrousselGit.
